gdnative-derive/src/native_script.rs

```diff
@@ -36,12 +36,40 @@ pub(crate) fn derive_native_class(input: TokenStream) -> TokenStream {
                 None
             };
 
+            let get_hook = if let Some(get_hook_expr) = config.get_hook {
+                let get_hook: ExprCall = parse_quote!(#get_hook_expr(this, _owner));
+                Some(get_hook)
+            } else {
+                None
+            };
```

Comment:
What's stopping you from using `Option::map`?

Reply:
> `set_before_hook` and `set_after_hook` are ambiguous since they can be interpreted as the actions of "setting the (unspecified) before/after hook". It is possible for a get hook to mutate the script struct using interior mutability, and we can also generate `&mut` getters instead if get hooks are present, so I'd say that an `after_get` hook might still be useful.

I will add back the before/after get hooks and rename everything appropriately. I am not going to be working on getting doing the mut getters. That seems like a separate change. Unfortunately, I don't have time to grapple with that one for now.

Reply:
This part is done. Working on other comments.

Reply:
This commend landed in a weird place. Let me fix this.

Reply:
My inexperience with rust. I'm working on this right now.

Reply:
Now fixed.